rfcs/text/0000-data-processing-robustness.md

```diff
@@ -0,0 +1,75 @@
+
+To create a robust and resilient system, it is critical that DCP components are engineered with a liberal attitude to data and experimental metadata, being highly tolerant to errors in data and metadata, integrity problems, and unexpected content. This is likely to mean that DCP components encounter data that they cannot handle, and this is an acceptable compromise as long as these cases are highly visible. This RFC proposes several strategies to engineer data processing systems for robustness and data integrity.
+
+### User Stories
+
+- As a DCP developer, I don't want to make urgent fixes to software due to differences or errors in data, so that development efforts can be planned and managed in a controlled manner, rather than a reactive one.
+- As a DCP submitter, I want my data to be ingested promptly, without waiting for DCP developers to make modifications to handle my experiment, so that my lab can move on to other tasks.
+- As a DCP data consumer, I don't want to receive scientifically incorrect, inaccurate or incomplete data caused by developers attempting to work around problematic data, so that I can spend my time on actual research.
+
+## Detailed Design
+
+In this RFC, we require two main strategies for making changes across the DCP that will ensure greater reliability and predictability of data processing. These strategies will reduce coupling between components, providing the opportunity for development teams to react in a controlled, planned manner whenever data is encountered that fails to meet the assumptions or expectations of DCP software components.
+
+The proposed strategies are:
+1. *Log and Continue*
+2. *Repair and Recover*
+
+### Log and Continue
+
+- DCP components define their expectations of any data they receive (see Arathi's metadata RFC)
+- DCP components ignore data they retrieve if it is mismatched against their expectations (e.g. bundles contain unrecognised file formats such as PDFs)
+- DCP components skip processing of data and fail gracefully when they encounter an error, rather than crashing out.
+- DCP components report correct error statuses (e.g. via HTTP response codes) to dependent
+
+
+### Repair and Recover
+
+- DCP components include operator admin functions to manually trigger automated steps that may have failed
+- In case of errors that causes some data to be skipped, DCP components ensure that all data expected to be handled together (e.g. all data from a single project) complete before making some of the data available
+
+
+### Unresolved Questions
+
+- It is expected that this RFC to raise questions that require more details specification of the requirements.
+- The most each component determine and reports compliance to these principles need to be defined.
+
```

Comment:
How each component determine**s** and reports compliance to these principles need**s** to be defined.